src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2638,6 +2638,65 @@ public static void UsingAlternativeTimeZoneIdsTest(string windowsId, string iana
+            }
+
+            Assert.False(TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time").HasIanaId, $"`Pacific Standard Time` should not be IANA Id.");
+            Assert.True(TimeZoneInfo.FindSystemTimeZoneById("America/Los_Angeles").HasIanaId, $"'America/Los_Angeles' should be IANA Id");
+        }
+
+        [ConditionalTheory(nameof(SupportIanaNamesConversion))]
+        [InlineData("Pacific Standard Time", "America/Los_Angeles")]
+        [InlineData("AUS Eastern Standard Time", "Australia/Sydney")]
+        [InlineData("GMT Standard Time", "Europe/London")]
+        [InlineData("Tonga Standard Time", "Pacific/Tongatapu")]
+        [InlineData("W. Australia Standard Time", "Australia/Perth")]
+        [InlineData("E. South America Standard Time", "America/Sao_Paulo")]
+        [InlineData("E. Africa Standard Time", "Africa/Nairobi")]
+        [InlineData("W. Europe Standard Time", "Europe/Berlin")]
+        [InlineData("Russian Standard Time", "Europe/Moscow")]
+        [InlineData("Libya Standard Time", "Africa/Tripoli")]
+        [InlineData("South Africa Standard Time", "Africa/Johannesburg")]
+        [InlineData("Morocco Standard Time", "Africa/Casablanca")]
+        [InlineData("Argentina Standard Time", "America/Buenos_Aires")]
+        [InlineData("Newfoundland Standard Time", "America/St_Johns")]
+        [InlineData("Iran Standard Time", "Asia/Tehran")]
+        public static void IdsConversionsTest(string windowsId, string ianaId)
+        {
+            Assert.True(TimeZoneInfo.TryConvertIanaIdToWindowsId(ianaId, out string winId));
+            Assert.Equal(windowsId, winId);
+
+            Assert.True(TimeZoneInfo.TryConvertWindowsIdToIanaId(winId, out string ianaConvertedId));
+            Assert.Equal(ianaId, ianaConvertedId);
+        }
+
+        [ConditionalTheory(nameof(SupportIanaNamesConversion))]
+        [InlineData("Pacific Standard Time", "America/Vancouver", "CA")]
+        [InlineData("Pacific Standard Time", "America/Los_Angeles", "US")]
+        [InlineData("Pacific Standard Time", "America/Los_Angeles", "\u0600NotValidRegion")]
+        [InlineData("Central Europe Standard Time", "Europe/Budapest", "DJ")]
+        [InlineData("Central Europe Standard Time", "Europe/Budapest", "\uFFFFNotValidRegion")]
+        [InlineData("Central Europe Standard Time", "Europe/Prague", "CZ")]
+        [InlineData("Central Europe Standard Time", "Europe/Ljubljana", "SI")]
+        [InlineData("Central Europe Standard Time", "Europe/Bratislava", "SK")]
+        [InlineData("Central Europe Standard Time", "Europe/Tirane", "AL")]
+        [InlineData("Central Europe Standard Time", "Europe/Podgorica", "ME")]
+        [InlineData("Central Europe Standard Time", "Europe/Belgrade", "RS")]
+        public static void IdsConversionsWithRegionTest(string windowsId, string ianaId, string region)
+        {
+            Assert.True(TimeZoneInfo.TryConvertWindowsIdToIanaId(windowsId, region, out string ianaConvertedId));
```

Comment:
Were you going to add some "invalid" region tests? Just to make sure the API behaves correctly when unicode characters are passed into this string.

Reply:
Yes, I already did.

```
[InlineData("Pacific Standard Time", "America/Los_Angeles", "\u0600NotValidRegion")]
[InlineData("Central Europe Standard Time", "Europe/Budapest", "\uFFFFNotValidRegion")]
```